lldb/tools/lldb-dap/tool/lldb-dap.cpp

```diff
@@ -342,6 +342,7 @@ serveConnection(const Socket::SocketProtocol &protocol, const std::string &name,
                      << " disconnected failed: "
                      << llvm::toString(std::move(error)) << "\n";
       }
+      dap->StopEventHandlers();
       // Close the socket to ensure the DAP::Loop read finishes.
       sock->Close();
```

Comment:
We may need to remove this `sock->Close()`. It should be closed when the `DAP::Loop()` function returns on line 308. I think that might be related to the race happening on interrupt.